### DISCUSSION TITLE
Make bash script to update hash values inplace

### DISCUSSION THREAD

**Discussion by tejasbhosale009** [mikefarah/yq#1041] [Q&A]
Hi,
This is my hiera yaml example file which i want to update inplace

```
profile::antcandy::location:
  'abc.example.com':
    v1: "3001"
    v4: "3004"
```
created a bash script to update this file inplace

```
#!/bin/bash

export myenvdata='"v1": "3001", "v2": "3006"'


echo "$myenvdata"

yq eval -i -v -I 5 '."profile::antcandy::location"."abc.example.com" |= env(myenvdata)' /root/devapi01.example.com.yaml

```

but this does not work ..

but if i use that command manually from command line bash shell then update works fine

```yq eval -i -v -I 5 '."profile::antcandy::location"."abc.example.com" |= {"v1": "3001", "v2": "3006"} /root/devapi01.example.com.yaml```

Can anyone please guide ??

**Comment by mikefarah** [comment]
You just need to include the `{}` in the environment value, like you do when you did it manually:

```

export myenvdata='{"v1": "3001", "v2": "3006"}'
echo "$myenvdata"

yq eval -i -v -I 5 '."profile::antcandy::location"."abc.example.com" |= env(myenvdata)' /root/devapi01.example.com.yaml

```

**Reply by rkarpuzov** [reply]
Hello,
I have similar case:

```
host_config:
  public.example.org:
    sql_type: pgsql
    sql_server: localhost
```

Where I want to store the domain name in env. How to use it in this case.

And other question is when I use the sql part as mentioned above, I receive the following syntax:

```
#executed in bash:
export pgsql_connect='{"sql_type": "pgsql", "sql_server":"localhost"}'
./yq '.host_config."public.example.org" |= env(pgsql_connect)' ejabberd.yml
```

```
 host_config:
      public.example.org: {"sql_type": "pgsql", "sql_server": "localhost"}
```
which syntax is not expected. 

Best regards,
Rossen

**Comment by mikefarah** [comment]
First problem:
Extract the value and assign to a variable in bash:
```
export hostName = $(yq '.host_config."public.example.org".sql_server' file.yaml)
```

Second:
Looks like the command worked, its just using a different yaml style (JSON is a subset of YAML spec). To get idiomatic yaml, you can recursively reset the style:

```
./yq 'with(.host_config."public.example.org"; . |= env(pgsql_connect) | ... style="" )' ejabberd.yml
```

**Reply by rkarpuzov** [reply]
My idea is not to export, but to store within the yq. Something like: 
export myhostname="public.example.org"
./yq 'with(.host_config; . |= env(myhostname); . |= env(pgsql_connect) | ... style="")' ejabberd.yml

the style works now as expected. Thanks!
